File: Projekti/Hibernate-Benchmark/src/main/java/com/example/HibernateBenchmark/benchmark/Benchmark.java
package com.example.HibernateBenchmark.benchmark;

import com.example.HibernateBenchmark.model.User;
import com.example.HibernateBenchmark.repository.PostRepository;
import com.example.HibernateBenchmark.repository.ProfileRepository;
import com.example.HibernateBenchmark.repository.TagRepository;
import com.example.HibernateBenchmark.repository.UserRepository;
import com.example.HibernateBenchmark.util.ResultWriter;
import com.example.HibernateBenchmark.util.Stopwatch;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.transaction.interceptor.TransactionAspectSupport;

import java.time.LocalDate;

@RequiredArgsConstructor
@Service
public class Benchmark {

    private final UserRepository userRepository;

    private final ProfileRepository profileRepository;

    private final PostRepository postRepository;

    private final TagRepository tagRepository;

    private final Stopwatch stopwatch = new Stopwatch();

    public void runAll(int times) {
        executeFirstQuery(times);
        //executeSecondQuery(times);
        //executeThirdQuery(times);
        //executeFourthQuery(times);
        //executeFifthQuery(times);
        //executeSixthQuery(times);
        //executeSeventhQuery(times);
        //executeEightQuery(times);
        //executeNinthQuery(times);
        //executeTenthQuery(times);
    }

    public void executeFirstQuery(int times){

        ResultWriter.writeHeader("Q1");

        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var users = userRepository.findByDateOfBirthAfter(LocalDate.of(2000,12,31));

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeSecondQuery(int times) {

        ResultWriter.writeHeader("Q2");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var users = userRepository.findUsersWithPostCount();

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeThirdQuery(int times) {

        ResultWriter.writeHeader("Q3");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var users = userRepository.findTopPosters();

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeFourthQuery(int times) {

        ResultWriter.writeHeader("Q4");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var users = userRepository.findUsersByBioKeyword("59");

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeFifthQuery(int times) {

        ResultWriter.writeHeader("Q5");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            User newUser = new User();
            newUser.setFirstName("Peroni");
            newUser.setLastName("Peric");
            newUser.setEmail("pera@examplic.com");
            newUser.setDateOfBirth(LocalDate.of(2000, 12, 20));

            userRepository.save(newUser);

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeSixthQuery(int times) {

        ResultWriter.writeHeader("Q6");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var tags = tagRepository.countPostsPerTag();

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeSeventhQuery(int times) {

        ResultWriter.writeHeader("Q7");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var tags = tagRepository.countTagsPerUser();

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeEightQuery(int times) {

        ResultWriter.writeHeader("Q8");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var tags = tagRepository.findDistinctTagsByEmail("user199@example.com");

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }


    public void executeNinthQuery(int times) {

        ResultWriter.writeHeader("Q9");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            var tags = postRepository.updatePostContentForUser("TEXT", 99);

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }

    public void executeTenthQuery(int times) {
        ResultWriter.writeHeader("Q10");
        long totalMilliseconds = 0;

        for (int i = 0; i < times; i++) {
            stopwatch.start();

            profileRepository.deleteProfilesByPhonePrefix("58");

            stopwatch.stop();

            totalMilliseconds += stopwatch.getElapsedMilliseconds();
        }

        double average = (double) totalMilliseconds / times;
        ResultWriter.writeAverageFooter(average);
    }
}

File: Projekti/Python-Native-Benchmark/benchmark/benchmark.py
from util.stopwatch import Stopwatch
from util.result_writer import print_footer, print_header, print_average_footer
from repository.post_repository import PostRepository
from repository.user_repository import UserRepository
from repository.tag_repository import TagRepository
from repository.profile_repository import ProfileRepository
import datetime


class Benchmark:

    def __init__(self, connection):
        self.user_repository = UserRepository(connection)
        self.post_repository = PostRepository(connection)
        self.tag_repository = TagRepository(connection)
        self.profile_repository = ProfileRepository(connection)
        self.stopwatch = Stopwatch()

    def run_all(self, times):
        # self.execute_first_query(times)
        # self.execute_second_query(times)
        # self.execute_third_query(times)
        # self.execute_fourth_query(times)
        #self.execute_fifth_query(times)
        #self.execute_sixth_query(times)
        #self.execute_seventh_query(times)
        self.execute_eigth_query(times)
        self.execute_ninth_query(times)
        self.execute_tenth_query(times)

    def execute_first_query(self, times):

        print_header("Q1")

        totalMilliseconds = 0

        year = datetime.date(2000, 1, 1)

        for i in range (1, times):

            self.stopwatch.start()

            users = self.user_repository.get_users_born_after(year)

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_second_query(self, times):

        print_header("Q2")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.user_repository.get_number_of_posts_for_each_user()

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_third_query(self, times):

        print_header("Q3")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.user_repository.get_top_five_users()

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_fourth_query(self, times):

        print_header("Q4")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.user_repository.get_users_by_keyword("59")

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_fifth_query(self, times):

        print_header("Q5")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            self.user_repository.add_user("Pera", "Petrovic", "perica@example.com", datetime.date(2000, 1, 1))

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_sixth_query(self, times):

        print_header("Q6")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.tag_repository.count_posts_per_tag()

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_seventh_query(self, times):

        print_header("Q7")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.tag_repository.count_tags_per_user()

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_eigth_query(self, times):

        print_header("Q8")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.tag_repository.get_tags_by_user("user199@example.com")

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_ninth_query(self, times):

        print_header("Q9")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.post_repository.update_posts_for_user(99, "TEXT")

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)

    def execute_tenth_query(self, times):

        print_header("Q10")

        totalMilliseconds = 0

        for i in range(1, times):

            self.stopwatch.start()

            result = self.profile_repository.delete_profiles_by_phone_number("87")

            self.stopwatch.stop()

            totalMilliseconds += self.stopwatch.get_milliseconds()

        average = totalMilliseconds*1.0 / times
        print_average_footer(average)
